src/rebar_hex_repos.erl

```diff
@@ -105,6 +105,15 @@ update_organizations(Repos) ->
                       Repo
               end, Repos).
 
+maybe_apply_mirror(Repos, State) ->
```

Comment:
This here is a good change and the only way it would make sense to have this override, +1 there on the logic.

However I don't think we should call this `mirror`; the mirror configuration is ideally done by using what is documented at http://rebar3.org/docs/configuration/configuration/#hex-repos-and-indexes -- the reason for this is that each mirror (especially non-official ones) may have out-of-date information, and the layered configuration in the rebar config allows to query multiple mirrors at once, and even allowing to have partial mirrors with distinct signing keys.

This function is strictly `maybe_override_default_repo_url()` -- and as I have mentioned in earlier comments in this PR, `HEX_CDN` isn't a very good name anymore for what it is supposed to do. At least giving the function a clear name will make it more obvious how limited its scope should be.